core/src/main/java/org/ldaptive/SearchRequest.java

```diff
@@ -29,7 +30,7 @@
   private SearchScope searchScope = SearchScope.SUBTREE;
 
   /** Time search operation will block. */
-  private long timeLimit;
+  private Duration timeLimit = Duration.ZERO;
```

Comment:
Is there a reason this default (ZERO) is different from those in ConnectionConfig (null)? Seems you'd want consistent defaults in all cases.